### DISCUSSION TITLE
How to determine if the intersection point falls on the emitter

### DISCUSSION THREAD

**Discussion by ioissss** [mitsuba-renderer/mitsuba3#1208] [Q&A] [answered]
After getting the scene intersection points via `scene.ray_intersect()` , how to determine whether these points are located on the emitter or not?

**Comment by njroussel** [comment, marked as answer]
Hi @ioissss 

There's a method in the `SurfaceInteraction3f` struct for this: https://mitsuba.readthedocs.io/en/latest/src/api_reference.html#mitsuba.SurfaceInteraction3f.emitter2
You can then do `mask = dr.eq(emitter, None)` to get a boolean mask.

It takes in the `scene` as a parameter so that it can return the envrionment map if the ray didn't hit any surface.
